### DISCUSSION TITLE
Elastic Search / Mongo DB - Where does this fit in

### DISCUSSION THREAD

**Discussion by andrew1555** [jasontaylordev/CleanArchitecture#493] [General]
Hi

If we need to use elastic search,  mongo db etc. where would we put the entity objects custom to the documents.  It seems domain typically matches SQL tables and if we added into the domain would mean coupling domain to different libraries.

I was thinking to abstract into the infrastructure layer and create a repository like _elasticsearch.getproduct() or something but then is this bad adding an extra layer?

Thanks

**Comment by karimkod** [comment]
Hi, 

I think this one is a good resource covering what you are refering too : https://docs.microsoft.com/en-us/dotnet/architecture/microservices/microservice-ddd-cqrs-patterns/nosql-database-persistence-infrastructure . 

Repository is also a good pattern to go with. 

Good luck!
